pyomo/contrib/mindtpy/iterate.py

```diff
@@ -97,25 +98,80 @@ def MindtPy_iteration_loop(solve_data, config):
-                config.call_after_subproblem_solve(fixed_nlp, solve_data)
+                # Call the NLP post-solve callback
+                with time_code(solve_data.timing, 'Call after subproblem solve'):
+                    config.call_after_subproblem_solve(fixed_nlp, solve_data)
 
-        if algorithm_should_terminate(solve_data, config, check_cycling=False):
-            last_iter_cuts = True
-            break
+                if algorithm_should_terminate(solve_data, config, check_cycling=False):
+                    last_iter_cuts = True
+                    break
+            else:
+                if config.mip_solver == 'cplex_persistent':
+                    solution_pool_names = main_mip_results._solver_model.solution.pool.get_names()
+                elif config.mip_solver == 'gurobi_persistent':
+                    solution_pool_names = list(
+                        range(main_mip_results._solver_model.SolCount))
+                solution_name_obj = []
+                for name in solution_pool_names:
+                    if config.mip_solver == 'cplex_persistent':
+                        obj = main_mip_results._solver_model.solution.pool.get_objective_value(
+                            name)
+                    elif config.mip_solver == 'gurobi_persistent':
+                        main_mip_results._solver_model.setParam(
+                            gurobipy.GRB.Param.SolutionNumber, name)
+                        obj = main_mip_results._solver_model.PoolObjVal
+                    solution_name_obj.append([name, obj])
+                solution_name_obj.sort(
+                    key=lambda x: x[1], reverse=solve_data.objective_sense == maximize)
+                counter = 0
+                for name, _ in solution_name_obj:
+                    # the optimal solution of the main problem has been added to integer_list above
+                    # so we should skip checking cycling for the first solution in the solution pool
+                    if counter >= 1:
+                        copy_var_list_values_from_solution_pool(solve_data.mip.MindtPy_utils.variable_list,
+                                                                solve_data.working_model.MindtPy_utils.variable_list,
+                                                                config, solver_model=main_mip_results._solver_model,
+                                                                var_map=main_mip_results._pyomo_var_to_solver_var_map,
+                                                                solution_name=name)
+                        solve_data.curr_int_sol = get_integer_solution(
+                            solve_data.working_model)
+                        if solve_data.curr_int_sol in set(solve_data.integer_list):
+                            config.logger.info(
+                                'The same combination has been explored and will be skipped here.')
```

Comment:
THis is just a question, wouldn't it be useful to tell the user which combinations have been previously explored? Maybe not under logger.info but under logger.debug

Reply:
I remember we tried to print the detailed combination if cycling happened (eg. [1, 0, 0, 1, 1, ...]). Actually, we cannot get any information from the numbers. So we choose to tell the user this combination has been explored in which previous iteration. I'll do so here.

Reply:
But since the solution pool is activated, we cannot easily index the store `integer_list` and get the iteration in which the same combination has been explored.